packages/application-shell/src/components/custom-view-shell/custom-view-shell.tsx

```diff
@@ -67,6 +67,10 @@ function StrictModeEnablement(props: TStrictModeEnablementProps) {
   }
 }
 
+const isLocalProdMode =
+  process.env.NODE_ENV === 'production' &&
+  window.location.origin.match(/http:\/\/(localhost|127.0.0.1)/g);
```

Comment:
I guess this is mainly so the template runs with the project code built for production but the development host is also available to be able to actually interact with the Custom View, right?

Instead of relying in the origin, maybe can add another env variable when building the template which can be used in this file to decide to render the mock host.

What do you think? 🤔

Reply:
Thank you 🙏, 

> I guess this is mainly so the template runs with the project code built for production but the development host is also available to be able to actually interact with the Custom View, right?

That's exactly it.

In terms of the suggested alternative, it seems theoretically valid and feasible.
It's worth noting that this approach would set a new precedent, as we typically do not inject environment variables directly into the JavaScript runtime, aside from `NODE_ENV`. However, I checked that and we can update Webpack and Vite production config to do that.

Reply:
There is already a variable for such use cases: `MC_APP_ENV`.

On runtime you can access that via `window.app.env`.